packages/auth/src/components/EmailPasswordSignUpForm/index.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Input } from 'react-rainbow-components';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { showAppMessage, showAppSpinner, hideAppSpinner } from '../../../../app/src/actions';
+import useFirebaseApp from '../../../../firebase-hooks/src/useFirebaseApp';
+
+const EmailPasswordSignUpForm = (props) => {
+    const intl = useIntl();
+    const { className, style } = props;
+    const [name, setName] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const app = useFirebaseApp();
+    const { nameIcon, emailIcon, passwordIcon } = props;
+
+    const onSubmit = async (event) => {
+        event.preventDefault();
+
+        try {
+            showAppSpinner();
+            await app.auth().createUserWithEmailAndPassword(email, password);
+            hideAppSpinner();
+        } catch (error) {
+            hideAppSpinner();
+            showAppMessage({
+                message: `${error}`,
+                variant: 'error',
+            });
+        }
+    };
+
+    const nameLabel = <FormattedMessage id="EmailPasswordSignUpForm.name" defaultMessage="Name" />;
+    const emailLabel = (
+        <FormattedMessage id="EmailPasswordSignUpForm.email" defaultMessage="Email Address" />
+    );
+    const passwordLabel = (
+        <FormattedMessage id="EmailPasswordSignUpForm.password" defaultMessage="Password" />
+    );
+    const buttonLabel = (
+        <FormattedMessage id="EmailPasswordSignUpForm.signup" defaultMessage="Create account" />
+    );
```

Comment:
this const can be outside the component to avoid recreation of them in each rerender